### PR TITLE
[Improve][Connector-v2][Jdbc]  check url not null throw friendly message

### PR DESCRIPTION
**Pull request by jackyyyyyssss**
## Purpose of this pull request
if jdbc source don't set url the matchingFactories first is DB2DialectFactory acceptsURL method throw npe  unfriendly so if url is null Don't run it afterwards
code modify message  friendly   #5094

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by TaoZex — APPROVED** [review]

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialectLoader.java [hunk 1]
```diff
@@ -46,6 +48,8 @@ private JdbcDialectLoader() {}
      * @return The loaded dialect.
      */
     public static JdbcDialect load(String url, String compatibleMode) {
+        Preconditions.checkNotNull(
+                url, "Miss <Source> Jdbc config url ! Please check the config file.");
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialectLoader.java]: Why check whether the parameter exists here?

Comment (jackyyyyyssss) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialectLoader.java]: There are three reasons for this
1.if the URL is empty, the following code will throw a null pointer exception. Running the following code is meaningless
2. Check in advance that the URL provides a customized exception message that is clearer than the original error prompt. The original code runs to the final List<JdbcDialectFactory>matchingFactories=
FoundFactories. stream(). filter (f ->f. acceptsURL (URL)). collect (Collectors. toList()); Will throw a null pointer exception and display the information of DB2, causing confusion for users  issues  #5094 
3. This is about the entry method for loading the JdbcDialect dialect. Personally, I feel that the starting check parameter of the method is more suitable

**Comment by liugddx** [comment]
Can be checked with this method, `CheckConfigUtil.checkAllExists`.
https://github.com/apache/seatunnel/blob/86b1b7e31a351dafc31404f1b3262f0bc0ee70be/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java#L96

**Comment by jackyyyyyssss** [comment]
> Can be checked with this method, `CheckConfigUtil.checkAllExists`.
> 
> https://github.com/apache/seatunnel/blob/86b1b7e31a351dafc31404f1b3262f0bc0ee70be/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java#L96

done PTAL

**Comment by liugddx** [comment]
Wait for CI.

**Review by liugddx — DISMISSED** [review]
+1

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java [hunk 2]
```diff
@@ -93,6 +98,19 @@ public String getPluginName() {
 
     @Override
     public void prepare(Config pluginConfig) throws PrepareFailException {
+        CheckResult checkResult =
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java]: This is a little weird. We use ReadonlyConfig for now, if some required config not be setted, we should use 
```java
ConfigValidator.of(config).validate(new JdbcSourceFactory().optionRule());
```
to check it.

Also this is a temporary solution. cc @hailin0 @EricJoy2048 @TyrantLucifer

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java]: This logic will invoke at factory. It's a best choice.

Comment (jackyyyyyssss) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java]: I found that KuduSource CassandraSource uses this method, or do you have any better suggestions for me to make some modifications    @Hisoka-X

Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java]: Please replace your code to 
```java
ConfigValidator.of(config).validate(new JdbcSourceFactory().optionRule());
```

Comment (jackyyyyyssss) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java]: done   PTAL @Hisoka-X

Comment (jackyyyyyssss) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java]: please help rebuild ci  https://github.com/apache/seatunnel/actions/workflows/backend.yml 502

**Review by Hisoka-X — APPROVED** [review]
The CI error not related to this PR. I believe.

**Review by liugddx — APPROVED** [review]
+1

**Pull request merged**